Improve unfriendly requirement that examples must be run in their package directory.

Currently, both examples require loading some assets from disk, e.g.:

```Go
// Load the texture
texture, err := newTexture("square.png")
if err != nil {
	panic(err)
}
```

https://github.com/go-gl/examples/blob/45f0380c29532eac123c04e239943a4125955f8a/glfw31-gl41core-cube/cube.go#L82-L85

This means they'll work only if the working directory is the same as the Go package, otherwise there'll be a panic like:

```
panic: open square.png: no such file or directory

(huge stack trace)
```

This can be improved in many ways, and it's worth doing because this knowledge is hard to have when first looking at the examples. It's not described in any README, and even if it were, it's just unfriendly design.

### Simple Enhancement

The `panic(err)` can be replaced with a more helpful error message, e.g.:

```Go
if err != nil {
	log.Fatalln("must be running in directory where the Go package is:", err)
}
```

And some notes can be added to README.

### Advanced Enhancement

Assuming a proper Go installation and the source present in `$GOPATH`, it's possible to make the binary auto-detect where the assets folder is, and use that. For example:

```Go
func importPathToDir(importPath string) string {
	p, err := build.Import(importPath, "", build.FindOnly)
	if err != nil {
		log.Fatalln(err)
	}
	return p.Dir
}

var baseDir = importPathToDir("github.com/go-gl/examples/glfw31-gl41core-cube")

texture, err := newTexture(filepath.Join(baseDir, "square.png"))
```

This issue is extracted from https://github.com/go-gl/examples/issues/49#issuecomment-183190427.

--- a/glfw31-gl21-cube/cube.go
+++ b/glfw31-gl21-cube/cube.go
@@ -6,6 +6,7 @@
 package main
 
 import (
+	"go/build"
 	"image"
 	"image/draw"
 	_ "image/png"
@@ -61,7 +62,7 @@ func main() {
 func newTexture(file string) uint32 {
 	imgFile, err := os.Open(file)
 	if err != nil {
-		panic(err)
+		log.Fatalf("texture %q not found on disk: %v\n", file, err)
 	}
 	img, _, err := image.Decode(imgFile)
 	if err != nil {
@@ -201,4 +202,27 @@ func drawScene() {
 	gl.Vertex3f(-1, 1, -1)
 
 	gl.End()
+}
+
+// Set the working directory to the root of Go package, so that its assets can be accessed.
+func init() {
+	dir, err := importPathToDir("github.com/go-gl/examples/glfw31-gl21-cube")
+	if err != nil {
+		log.Fatalln("Unable to find Go package in your GOPATH, it's needed to load assets:", err)
+	}
+	err = os.Chdir(dir)
+	if err != nil {
+		log.Panicln("os.Chdir:", err)
+	}
+}
+
+// importPathToDir resolves the absolute path from importPath.
+// There doesn't need to be a valid Go package inside that import path,
+// but the directory must exist.
+func importPathToDir(importPath string) (string, error) {
+	p, err := build.Import(importPath, "", build.FindOnly)
+	if err != nil {
+		return "", err
+	}
+	return p.Dir, nil
 }

--- a/glfw31-gl41core-cube/cube.go
+++ b/glfw31-gl41core-cube/cube.go
@@ -7,6 +7,7 @@ package main
 
 import (
 	"fmt"
+	"go/build"
 	"image"
 	"image/draw"
 	_ "image/png"
@@ -81,7 +82,7 @@ func main() {
 	// Load the texture
 	texture, err := newTexture("square.png")
 	if err != nil {
-		panic(err)
+		log.Fatalln(err)
 	}
 
 	// Configure the vertex data
@@ -198,7 +199,7 @@ func compileShader(source string, shaderType uint32) (uint32, error) {
 func newTexture(file string) (uint32, error) {
 	imgFile, err := os.Open(file)
 	if err != nil {
-		return 0, err
+		return 0, fmt.Errorf("texture %q not found on disk: %v", file, err)
 	}
 	img, _, err := image.Decode(imgFile)
 	if err != nil {
@@ -314,4 +315,27 @@ var cubeVertices = []float32{
 	1.0, -1.0, 1.0, 1.0, 1.0,
 	1.0, 1.0, -1.0, 0.0, 0.0,
 	1.0, 1.0, 1.0, 0.0, 1.0,
+}
+
+// Set the working directory to the root of Go package, so that its assets can be accessed.
+func init() {
+	dir, err := importPathToDir("github.com/go-gl/examples/glfw31-gl41core-cube")
+	if err != nil {
+		log.Fatalln("Unable to find Go package in your GOPATH, it's needed to load assets:", err)
+	}
+	err = os.Chdir(dir)
+	if err != nil {
+		log.Panicln("os.Chdir:", err)
+	}
+}
+
+// importPathToDir resolves the absolute path from importPath.
+// There doesn't need to be a valid Go package inside that import path,
+// but the directory must exist.
+func importPathToDir(importPath string) (string, error) {
+	p, err := build.Import(importPath, "", build.FindOnly)
+	if err != nil {
+		return "", err
+	}
+	return p.Dir, nil
 }

